src/Resources/views/CRUD/list__action_delete.html.twig

```diff
@@ -10,12 +10,21 @@ file that was distributed with this source code.
 #}
 
 {% if admin.hasAccess('delete', object) and admin.hasRoute('delete') %}
+    {% set button_content = sonata_config.getOption('list_action_button_content') %}
     <a
         href="{{ admin.generateObjectUrl('delete', object, actions.link_parameters|default([])) }}"
         class="btn btn-sm btn-default delete_link"
         title="{{ 'action_delete'|trans({}, 'SonataAdminBundle') }}"
     >
-        <i class="fa fa-times" aria-hidden="true"></i>
+        {% if button_content == 'icon' or button_content == 'all' %}
+            <i class="fa fa-times" aria-hidden="true"></i>
+        {% endif %}
+        {% if button_content != 'text' and button_content != 'all' %}
```

Comment:
Maybe we could set the result of this condition in a variable to reuse the same logic for the closing `</span>`. We can also use a `if`/`else` condition.
Personally, I don't feel comfortable redefining the same logic twice.